src/main/java/jenkins/branch/MultiBranchProjectViewHolder.java

```diff
@@ -70,11 +71,11 @@
      * The list of {@link View}s.
      */
     @GuardedBy("this")
-    private transient List<View> views = null;
+    private transient volatile List<View> views = null;
     /**
      * The primary view name.
      */
-    @GuardedBy("this")
+    @SuppressFBWarnings(value = "IS2_INCONSISTENT_SYNC", justification = "Only need to synchronize initialization")
     private transient String primaryView = null;
```

Comment:
Not 100% on if this should be volatile or not

Reply:
Yeah, I'm not really sure if it's necessary in this case. I would probably make `primaryView` `volatile` for now to be safe, since there should still be a performance improvement because of the reduced locking, and if anyone wants to investigate the semantics in detail in the future in case `primaryView` doesn't need to be `volatile`, they can do so.

I guess the easier way to reason about it would be to put `views` and `primaryView` in a composite object that holds them via `final` fields instead of having them be directly in `MultiBranchProjectViewHolder` (at that point the holder might not even need `volatile` because it would be immutable?), which would make it a more typical case of double-checked locking instead of having two fields interact with the lock.